improver_tests/utilities/test_neighbourhood_tools.py

```diff
@@ -87,3 +97,108 @@ def test_rolling_window_writable(array_size_5):
     windows = rolling_window(array_size_5, (2, 2), writeable=True)
     windows[0, 0, 0, 0] = -1
     assert windows[0, 0, 0, 0] == -1
+
+
+def test_padding_neighbourhood_size_2(array_size_5):
+    """Test that result is same as result of rolling_window with a border of zeros."""
+    padded = pad_and_roll(array_size_5, (2, 2), mode="constant")
+    window = rolling_window(array_size_5, (2, 2))
+    inner_part = padded[1:-1, 1:-1, ::]
+    np.testing.assert_array_equal(inner_part, window)
+    border_index = (
+        [[0, i, 0, j] for i in range(5) for j in [0, 1]]
+        + [[5, i, 1, j] for i in range(5) for j in [0, 1]]
+        + [[i, 0, j, 0] for i in range(5) for j in [0, 1]]
+        + [[i, 5, j, 1] for i in range(5) for j in [0, 1]]
+    )
+    outer_part = padded[list(zip(*border_index))]
+    np.testing.assert_array_equal(outer_part, np.zeros(40, dtype=np.int32))
+
+
+def test_padding_non_zero(array_size_5):
+    """Test padding with a number other than the default of 0."""
+    padded = pad_and_roll(array_size_5, (2, 2), mode="constant", constant_values=1)
+    border_index = (
+        [[0, i, 0, j] for i in range(5) for j in [0, 1]]
+        + [[5, i, 1, j] for i in range(5) for j in [0, 1]]
+        + [[i, 0, j, 0] for i in range(5) for j in [0, 1]]
+        + [[i, 5, j, 1] for i in range(5) for j in [0, 1]]
+    )
+    outer_part = padded[list(zip(*border_index))]
+    np.testing.assert_array_equal(outer_part, np.ones(40, dtype=np.int32))
+
+
+def test_pad_boxsum(array_size_3):
+    """Test that padded array consists of input array surrounded by border of zeros."""
+    padded = pad_boxsum(array_size_3, 3, mode="constant")
+    expected = np.zeros((6, 6), dtype=np.int32)
+    expected[2:5, 2:5] = array_size_3
+    np.testing.assert_array_equal(padded, expected)
+
+
+def test_pad_boxsum_non_zero(array_size_3):
+    """Test padding with a number other than the default of 0."""
+    padded = pad_boxsum(array_size_3, 3, mode="constant", constant_values=2)
+    expected = 2 * np.ones((6, 6), dtype=np.int32)
+    expected[2:5, 2:5] = array_size_3
+    np.testing.assert_array_equal(padded, expected)
+
+
+def test_boxsum_no_cumsum(array_size_5):
```

Comment:
Perhaps the name of this test is a bit confusing given that the `no_cumsum` test is when the `cumsum` argument is set to true? Maybe it would be better to change the names of `test_boxsum_no_cumsum` and `test_boxsum_with_cumsum`, so perhaps something like: `test_boxsum_automatic_cumsum` and `test_boxsum_specify_cumsum`, or something like that.